src/evaluating_students/router.js

```diff
@@ -1,14 +1,360 @@
+      })
+      .catch(() => res.sendStatus(500))
+  );
+
+/**
+ * @swagger
+ *  paths:
+ *      /tests:
+ *         get:
+ *           summary: Return list of all tests
+ *           responses:
+ *              '200':
+ *                  description: "List of all tests"
+ *              '500':
+ *                  description: "Database query failed"
+ *         put:
+ *           summary: Not implemented.
+ *           responses:
+ *              '501':
+ *                  description: "Not implemented"
+ *         post:
+ *           summary: Add test to test collection
+ *           parameters:
+ *              -   in: body
+ *                  name: test
+ *                  description: The test to create.
+ *                  schema:
+ *                      type: object
+ *                      required:
+ *                          -   course
+ *                          -   date
+ *                      properties:
+ *                          course:
+ *                              type: string
+ *                          date:
+ *                              type: date
+ *           responses:
+ *              '201':
+ *                  description: "[Saved document test]"
+ *              '400':
+ *                  description: "Bad request"
+ *              '500':
+ *                  description: "Failed to save document"
+ *         delete:
+ *           summary: Deletes all documents in the test collection
+ *           responses:
+ *              '200':
+ *                  description: "[Count of deleted documents]"
+ *              '500':
+ *                  description: "Failed to delete document"
+ */
+router
+  .route("/tests")
+  .get((req, res) => {
+    TestModel.find({})
+      .then(alltests => res.json(alltests))
+      .catch(() => res.sendStatus(500));
+  })
+
+  .put((req, res) => res.sendStatus(501))
+
+  .post((req, res) => {
+    let test = new TestModel();
+
+    const course = req.body.course;
+    const date = req.body.date;
+
+    if (course === undefined || date === undefined) {
+      res.sendStatus(400);
+      return;
+    }
+
+    test.course = req.body.course;
+    test.testDate = req.body.date;
+    test
+      .save()
+      .then(savedtest => {
+        res.status(201);
+        res.json(savedtest);
+      })
+      .catch(err => res.sendStatus(500));
+  })
+
+  .delete((req, res) =>
+    TestModel.remove({})
+      .then(deleted => {
+        res.status(200);
+        res.json(deleted);
+      })
+      .catch(() => res.sendStatus(500))
+  );
+
+router.route("/tests/:id/enroll-student").post((req, res) => {
+  const course = req.params.id;
+  const firstName = req.body.firstName;
+  const lastName = req.body.lastName;
+
+  if (
+    course === undefined ||
+    firstName === undefined ||
+    lastName === undefined
+  ) {
+    res.sendStatus(400);
+    return;
+  }
+
+  TestModel.findById(course).then(test => {
+    test.enrolledStudents.push({ firstName: firstName, lastName: lastName });
```

Comment:
`{firstName: firstName, lastName: lastName}`  is overbodig, je kunt het veranderen naar `{ firstName, lastName }`